tools/delete_mri_upload.pl

```diff
@@ -0,0 +1,578 @@
+	exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+if(!defined $tarchiveLibraryDir && $archiveLocation !~ /^\//) {
+    print STDERR "Cannot determine absolute path for archive '$archiveLocation' "
+        . "since config setting 'tarchiveLibraryDir' is not set. Aborting.\n";
+    exit $NeuroDB::ExitCodes::MISSING_CONFIG_SETTING;
+}
+
+$archiveLocation = "$tarchiveLibraryDir/$archiveLocation" unless $archiveLocation =~ /^\//;
+
+#===================================================#
+# Make sure there are no entries in files_qc_status #
+# and feedback_mri_comments for the files in the    #
+# archive                                           #
+#===================================================#
+
+if(&hasQcOrComment($dbh, $uploadID, $tarchiveID)) {
+	print STDERR "Cannot delete upload '$uploadID': there is QC information"
+        . " associated to that archive\n";
+	exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+#===========================================#
+# Backup all files found in                 #
+# tables files and files_intermediary for   #
+# the specified upload_id                   #
+#===========================================#
+
+my $filesRef             = &getFilesRef($dbh, $tarchiveID, $dataDirBasepath);
+my $intermediaryFilesRef = &getIntermediaryFilesRef($dbh, $tarchiveID, $dataDirBasepath);
+my $picFilesRef          = &getPicFilesRef($dbh, $tarchiveID, $dataDirBasepath);
+
+unless($noBackup) {
+    &backupFiles($uploadID, $archiveLocation, $filesRef, $intermediaryFilesRef, $picFilesRef);
+}
+&deleteUploadInDatabase($dbh, $uploadID, $tarchiveID, $sessionID, $intermediaryFilesRef, $picFilesRef);
+&deleteUploadFiles($archiveLocation, $filesRef, $intermediaryFilesRef, $picFilesRef);
+
+print "Upload $uploadID successfully deleted.\n";
+exit $NeuroDB::ExitCodes::SUCCESS;
+
+=pod
+
+=head3 hasQcOrComment($dbh, $tarchiveID)
+
+Determines if a tarchive has QC information associated to it by looking at the
+contents of tables C<files_qcstatus> and C<feedback_mri_comments>.
+
+INPUTS:
+
+  - $dbhr  : database handle reference.
+  - $tarchiveID: ID of the tarchive.
+
+RETURNS:
+
+  1 if there is QC information associated to the archive, 0 otherwise.
+
+=cut
+sub hasQcOrComment {
+	my($dbh, $tarchiveID) = @_;
+	
+	#=========================================#
+    # Fetch contents of tables files_qcstatus #
+    # and feedback_mri_comments               #
+    #=========================================#
+    (my $query =<<QUERY) =~ s/\s+/ /g;
+         SELECT fqs.FileID
+         FROM files_qcstatus fqs 
+         JOIN files f USING (FileID)
+         WHERE f.TarchiveSource = ?
+         
+         UNION
+          
+         SELECT fmc.FileID
+         FROM feedback_mri_comments fmc 
+         JOIN files f USING (FileID)
+         WHERE f.TarchiveSource = ?
+         LIMIT 1
+QUERY
+
+    my $rowsRef = $dbh->selectall_arrayref($query, undef, $tarchiveID, $tarchiveID);
```

Comment:
the query should be prepared before execution

Reply:
not sure I understand the undef when executing the query. Looks like there is only 2 placeholders in the query, both for the tarchiveID.

Reply:
`selectall_arrayref` actually prepares the statement before executing it. The second parameter to this function is used to specify additional execution parameters (I do not need any, so I used `undef`).

See [https://metacpan.org/pod/DBI#selectall_arrayref](https://metacpan.org/pod/DBI#selectall_arrayref)

Reply:
OK. Perfect then :). Thanks for the link!